### PR TITLE
Retain id in output types and resource properties

### LINKED ISSUE #508: GroupKey Id missing from cloudidentity Group
### What happened?

The cloudidentity.Group response is missing the groupKey.id

### Steps to reproduce

1. Create a service account with a `Google Groups Admin Role` in Workspace, I followed: https://medium.com/google-cloud/create-google-groups-via-terraform-de99524f92e0

2. Create a new cloudIdentity group and see that it is missing the `.groupKey.id`:

```typescript
const group = new cloudidentity.v1.Group("foobar", {
  groupKey: {id: "foobar@example.com"},
  initialGroupConfig: "EMPTY",
  labels: {
    "cloudidentity.googleapis.com/groups.discussion_forum": "",
  },
  parent: `customers/${customerId}`
})
```

3. See that `group.groupKey.id` is `undefined` and does not compile.

### Expected Behavior

`group.groupKey.id` is an `Output(foobar@exmaple.com)`

### Actual Behavior

it's `undefined`

### Versions used

CLI
Version      3.33.2
Go Version   go1.17.10
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host
OS       ubuntu
Version  20.04
Arch     x86_64

This project is written in nodejs (/home/kaoru/.asdf/shims/node v16.14.2)

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by viveklak**
We were previously omitting all `id` fields from all response types in the fear of this conflicting with the `id` injected by Pulumi SDK. Instead, we are now allowing all nested response types to retain the `id` field. In the top-level resource, we now rename any `id` fields to be `<resourcename>Id`.

Fixes #508

### CODE REVIEW

**Review by lblackstone — APPROVED** [review]

File: provider/pkg/gen/schema.go [hunk 1]
```diff
@@ -1024,9 +1024,9 @@ func (g *packageGenerator) genProperties(typeName string, typeSchema *discovery.
 		if !isOutput && readOnly {
 			continue
 		}
-		if isOutput && name == "id" {
-			continue
-		}
+		//if isOutput && name == "id" {
+		//	continue
+		//}
```
Comment (mikhailshilkov) [inline provider/pkg/gen/schema.go:1029]: As briefly discussed in slack, I think we should keep doing this for resources but not types

File: sdk/dotnet/AppEngine/V1/App.cs [hunk 2]
```diff
@@ -75,6 +75,12 @@ public partial class App : Pulumi.CustomResource
         [Output("iap")]
         public Output<Outputs.IdentityAwareProxyResponse> Iap { get; private set; } = null!;
 
+        /// <summary>
+        /// Identifier of the Application resource. This identifier is equivalent to the project ID of the Google Cloud Platform project where you want to deploy your application. Example: myapp.
+        /// </summary>
+        [Output("id")]
+        public Output<string> Id { get; private set; } = null!;
```
Comment (mikhailshilkov) [inline sdk/dotnet/AppEngine/V1/App.cs:82]: You can see the warning from the C# compiler about `Id` overriding the base property incorrectly.

**Review by mikhailshilkov — CHANGES_REQUESTED** [review]

File: provider/pkg/gen/schema.go (hunk 1 above)
Comment (viveklak) [reply provider/pkg/gen/schema.go]: Thanks for the insight @mikhailshilkov. I have updated the PR now. Could you take another look?

**Comment by viveklak** [comment]
> ### Does the PR have any schema changes?
> Found 33 breaking changes: Resource "google-native:dataplex/v1:Entity" missing input "id" Resource "google-native:storage/v1:BucketAccessControl" missing input "id" Resource "google-native:appengine/v1beta:DomainMapping" ...

These are instances where the resource had a top level ID field which would conflict. These have been renamed to `<Resourcename>Id`.

**Comment by viveklak** [comment]
Updated the PR based on feedback from Mikhail.

File: provider/pkg/gen/schema.go [hunk 3]
```diff
@@ -528,13 +528,13 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 		// If the request type matches the pattern when it contains a property of the type equal to the response
 		// type of the GET endpoint, then we want to flatten that property, so that the resource inputs are a superset
 		// of the resource outputs. This also helps reconcile the shape of create and update operations.
-		var flatten string
+		flatten := codegen.NewStringSet()
```
Comment (mikhailshilkov) [inline provider/pkg/gen/schema.go:531]: Could you please explain this change and how it affects IDs?

File: provider/pkg/gen/schema.go [hunk 4]
```diff
@@ -594,7 +599,7 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 			}
 
 			for name, value := range updateBag.properties {
-				if _, has := bodyBag.properties[name]; has {
+				if _, has := resourceMeta.Create.SDKProperties[name]; has {
```
Comment (mikhailshilkov) [inline provider/pkg/gen/schema.go:602]: What does this change?

File: sdk/dotnet/Apigee/V1/Apicategory.cs [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+
+namespace Pulumi.GoogleNative.Apigee.V1
+{
+    /// <summary>
+    /// Creates a new category on the portal.
+    /// </summary>
+    [GoogleNativeResourceType("google-native:apigee/v1:Apicategory")]
+    public partial class Apicategory : Pulumi.CustomResource
```
Comment (mikhailshilkov) [inline sdk/dotnet/Apigee/V1/Apicategory.cs:16]: Do you know why this resource appeared?

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,7 +2,7 @@ CHANGELOG
 =========
 
 ## HEAD (Unreleased)
-(None)
+- Retain `id` fields in output types and resource properties [#521](https://github.com/pulumi/pulumi-google-native/pull/521)
```
Comment (mikhailshilkov) [inline CHANGELOG.md:5]: Do you want to highlight this as a breaking change for a number of resource input args?

**Review by mikhailshilkov — COMMENTED** [review]
This makes sense overall. I asked a few questions below

**Pull request closed without merging**